llvm/include/llvm/Analysis/IVDescriptors.h

```diff
@@ -47,6 +47,8 @@ enum class RecurKind {
   FMul,     ///< Product of floats.
   FMin,     ///< FP min implemented in terms of select(cmp()).
   FMax,     ///< FP max implemented in terms of select(cmp()).
+  FMaxNoFMFs, ///< FP max implemented in terms of select(cmp()), but without
+              ///any fast-math flags. Users need to handle NaNs and signed zeros when generating code.
```

Comment:
Rename to something like FCmpOGTSelect?

Reply:
Yeah, I originally named if FMaxNoFMFs in line with existing `FMax`, but the current version doesn't match all pattersn `FMax` currently does. Renamed as suggested, we can always rename once/if we reach parity

Reply:
This captures **a** potentially Unordered FMax pattern - potentially with NaN's and signed zeroes - but not **all** possible such FMax patterns, yet? If all other cases should be added to this RecurKind, then that may sound like the right name? Perhaps the existing FMax pattern should be denoted as "Ordered", as it relies on FMF to indicate absence of NaN's (i.e., all elements are "isOrdered()") and signed zeroes.